File: app/src/main/java/com/hitesh/todopod/itemsRepository.java
package com.hitesh.todopod;

import android.app.Application;
import android.os.AsyncTask;

import androidx.lifecycle.LiveData;

import java.util.List;

public class itemsRepository {
    private itemsDAO itemsDao;
    private LiveData<List<items>> allitemss;
    public itemsRepository(Application application) {
        itemsDatabase database = itemsDatabase.getInstance(application);
        itemsDao = database.itemsDAO();
        allitemss = itemsDao.getAllNotes();
    }
    public void insert(items items) {
        new InsertitemsAsyncTask(itemsDao).execute(items);
    }
    public void update(items items) {
        new UpdateitemsAsyncTask(itemsDao).execute(items);
    }
    public void delete(items items) {
        new DeleteitemsAsyncTask(itemsDao).execute(items);
    }
    public void deleteAllitemss() {
        new DeleteAllitemssAsyncTask(itemsDao).execute();
    }
    public LiveData<List<items>> getAllitemss() {
        return allitemss;
    }
    private static class InsertitemsAsyncTask extends AsyncTask<items, Void, Void> {
        private itemsDAO itemsDao;
        private InsertitemsAsyncTask(itemsDAO itemsDao) {
            this.itemsDao = itemsDao;
        }
        @Override
        protected Void doInBackground(items... itemss) {
            itemsDao.insert(itemss[0]);
            return null;
        }
    }
    private static class UpdateitemsAsyncTask extends AsyncTask<items, Void, Void> {
        private itemsDAO itemsDao;
        private UpdateitemsAsyncTask(itemsDAO itemsDao) {
            this.itemsDao = itemsDao;
        }
        @Override
        protected Void doInBackground(items... itemss) {
            itemsDao.update(itemss[0]);
            return null;
        }
    }
    private static class DeleteitemsAsyncTask extends AsyncTask<items, Void, Void> {
        private itemsDAO itemsDao;
        private DeleteitemsAsyncTask(itemsDAO itemsDao) {
            this.itemsDao = itemsDao;
        }
        @Override
        protected Void doInBackground(items... itemss) {
            itemsDao.delete(itemss[0]);
            return null;
        }
    }
    private static class DeleteAllitemssAsyncTask extends AsyncTask<Void, Void, Void> {
        private itemsDAO itemsDao;
        private DeleteAllitemssAsyncTask(itemsDAO itemsDao) {
            this.itemsDao = itemsDao;
        }
        @Override
        protected Void doInBackground(Void... voids) {
            itemsDao.deleteAllNotes();
            return null;
        }
    }
}

File: app/src/main/java/com/hitesh/todopod/itemsRepository.kt
package com.hitesh.todopod

import android.app.Application
import android.os.AsyncTask
import androidx.lifecycle.LiveData
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers.IO
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class itemsRepository {
    var itemsDao: itemsDAO
    var allitemss: LiveData<List<items>>

    constructor(application: Application?) {
        var database = itemsDatabase.getInstance(application)
        itemsDao = database.itemsDAO()
        allitemss = itemsDao.allNotes
    }

    fun insert(items: items?) {
        insertItemCoroutine(itemsDao, items) }

    fun update(items: items?) {
        updateItemCoroutine(itemsDao, items)
    }

    fun delete(items: items?) {
        DeleteItemsCoroutine(itemsDao, items)
    }

    fun deleteAllitemss() {
       DeleteAllItemsCoroutine(itemsDao)
    }

    fun insertItemCoroutine(itemsDao: itemsDAO?, itemss: items?){
      val itemsDao: itemsDAO? = null
        if (itemsDao != null) this.itemsDao = itemsDao
        
        CoroutineScope(IO).launch { insertItem(itemss) }
    }
    suspend fun insertItem(vararg itemss: items?): Void?{
        withContext(IO){
            itemsDao?.insert(itemss[0])
        }
        return null
    }

    fun updateItemCoroutine(itemsDao: itemsDAO?, itemss: items?){
        val itemsDao: itemsDAO? = null
        if (itemsDao != null) this.itemsDao = itemsDao

        CoroutineScope(IO).launch { updateItem(itemss) }
    }
    suspend fun updateItem(vararg itemss: items?): Void?{
        withContext(IO){
            itemsDao?.update(itemss[0])
        }
        return null
    }

    fun DeleteItemsCoroutine(itemsDao: itemsDAO?, itemss: items?){
        val itemsDao: itemsDAO? = null
        if (itemsDao != null) this.itemsDao = itemsDao

        CoroutineScope(IO).launch { deleteItem(itemss) }
    }
    suspend fun deleteItem(vararg itemss: items?): Void?{
        withContext(IO){
            itemsDao?.delete(itemss[0])
        }
        return null
    }

    fun DeleteAllItemsCoroutine(itemsDao: itemsDAO?){
        val itemsDao: itemsDAO? = null
        if (itemsDao != null) this.itemsDao = itemsDao

        CoroutineScope(IO).launch { deleteAllItem() }
    }
    suspend fun deleteAllItem(vararg itemss: items?): Void?{
        withContext(IO){
            itemsDao?.deleteAllNotes()
        }
        return null
    }
}